Downgrade Grep to 2.27-2 on windows

jdk8u builds require `grep 2.27-2` but cygwin defaults to `grep 3.0`. There seems to be no easy way to programatically downgrade the package so this may take a bit of working out but ideally needs to be added to the playbook. Installed manually using the cygwin.exe gui application for now.

error for reference:

```bash
make[2]: *** [/cygdrive/c/Users/jenkins/workspace/openjdk8_build_x86-64_windows/openjdk/build/windows-x86_64-normal-server-release/images/lib/_the.rt.jar.contents] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [images] Error 2```

--- a/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Firefox/tasks/main.yml
+++ b/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Firefox/tasks/main.yml
@@ -5,7 +5,7 @@
 
 - name: Test if Firefox is already installed
   win_stat:
-    path: 'C:\Program Files (x86)\Mozilla Firefox\firefox.exe'
+    path: 'C:\Program Files\Mozilla Firefox\firefox.exe'
   register: firefox_installed
   tags: Firefox
 

--- a/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml
+++ b/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml
@@ -55,6 +55,17 @@
     - xterm
   tags: cygwin
 
+- name: Download grep-2.27-2
+  win_get_url:
+    url: "{{ Cygwin_SITE_URL + '/x86_64/release/grep/grep-2.27-2.tar.xz' }}"
+    dest: "{{ Cygwin_ROOT_INST_DIR + '\\tmp\\grep.tar.xz' }}"
+    force: no
+  tags: cygwin
+
+- name: Extract grep-2.27-2
+  win_command: "{{ Cygwin_ROOT_INST_DIR + '\\bin\\bash.exe -l -c \"tar -C / -xvf /tmp/grep.tar.xz\"' }}"
+  tags: cygwin
+
 - name: Remove c:\cygwin64\bin from the path if it exists
   win_path:
     elements:

--- a/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml
+++ b/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml
@@ -55,6 +55,14 @@
     - xterm
   tags: cygwin
 
+###############################################################################
+# Cygwin by default installs latest versions of packages, which is 3.0-2 for
+# grep. This version of grep is unable to match EOL (with $) when the file use
+# windows style CRLF sequence at end of lines and it is required for openjdk8
+# builds to complete. The following two steps downloads the latest known
+# version of grep to satisfy this requirement and overwrites the installed
+# grep version.
+###############################################################################
 - name: Download grep-2.27-2
   win_get_url:
     url: "{{ Cygwin_SITE_URL + '/x86_64/release/grep/grep-2.27-2.tar.xz' }}"

